chat-android/src/main/java/com/ably/chat/Cancellation.kt

```diff
@@ -0,0 +1,12 @@
+package com.ably.chat
+
+/**
+ * A cancellation handle, returned by various functions (mostly subscriptions)
+ * where cancellation is required.
+ */
+fun interface Cancellation {
+    /**
+     * Handle cancellation (unsubscribe listeners, clean up)
+     */
+    fun cancel()
```

Comment:
In JS we use `unsubscribe` or `off` but happy to keep `cancel` if that's more kotlin-idiomatic.